src/components/SelectNumber/index.tsx

```diff
@@ -0,0 +1,354 @@
+  }
+
+  const handleChange: ChangeEventHandler<HTMLInputElement> = event => {
+    event.stopPropagation()
+    const parsedValue = parseIntOr(event.currentTarget.value, 0)
+    setValue(parsedValue)
+  }
+
+  const handleOnFocus: FocusEventHandler<HTMLInputElement> = event => {
+    if (onFocus) onFocus(event)
+  }
+
+  const handleOnBlur: FocusEventHandler<HTMLInputElement> = event => {
+    const boundedValue = bounded(inputValue, minValue, maxValue)
+
+    if (inputValue > maxValue) onMaxCrossed?.()
+    if (inputValue < minValue) onMinCrossed?.()
+
+    setValue(boundedValue)
+
+    if (onBlur) onBlur(event)
+  }
+
+  const onKeyDown: KeyboardEventHandler = e => {
+    // Arrow Up
+    if (e.keyCode === 38) {
+      e.stopPropagation()
+      e.preventDefault()
+
+      const direction = 1
+      const newValue =
+        inputValue % step === 0 ? inputValue + step * direction : inputValue
+      const roundedValue = roundStep(newValue, step, direction)
+
+      if (roundedValue <= maxValue) {
+        setValue(roundedValue)
+      }
+    }
+
+    // Arrow Down
+    if (e.keyCode === 40) {
+      e.stopPropagation()
+      e.preventDefault()
+
+      const direction = -1
+
+      const newValue =
+        inputValue % step === 0 ? inputValue + step * direction : inputValue
+      const roundedValue = roundStep(newValue, step, direction)
+
+      if (roundedValue >= minValue) {
+        setValue(roundedValue)
+      }
+    }
+  }
+
+  const minusRoundedValue =
+    inputValue % step === 0
+      ? roundStep(inputValue - step, step, -1)
+      : roundStep(inputValue, step, -1)
+  const plusRoundedValue =
+    inputValue % step === 0
+      ? roundStep(inputValue + step, step, 1)
+      : roundStep(inputValue, step, 1)
+  const isMinusDisabled = minusRoundedValue < minValue || disabled
+  const isPlusDisabled = plusRoundedValue > maxValue || disabled
+
+  return (
+    <StyledContainer disabled={disabled} size={size} {...props}>
+      <Tooltip text={isMinusDisabled && disabledTooltip}>
+        <StyledTouchable
+          onClick={offsetFn(-1)}
+          size={size}
+          disabled={isMinusDisabled}
+          aria-label="Minus"
+        >
+          <Icon name="minus" size={iconSizes[size]} />
+        </StyledTouchable>
+      </Tooltip>
+
+      <StyledCenterBox
+        size={size}
+        onClick={() => {
+          if (inputRef?.current) {
+            inputRef.current.focus()
+          }
+        }}
+        aria-live="assertive"
+        role="status"
+      >
+        <StyledInput
+          disabled={disabled}
+          name={name}
+          onBlur={handleOnBlur}
+          onChange={handleChange}
+          onFocus={handleOnFocus}
+          onKeyDown={onKeyDown}
+          ref={inputRef}
+          style={{
+            width: inputValue.toString().length * 10 + 15,
+          }}
+          value={inputValue.toString()} // A dom element can only have string attributes.
+          aria-label="Input"
```

Comment:
And bread is made of bread 😄 
It's better to put nothing if we can't have a relevant label :)